lk2nd/device/leds.c

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2025 Yang Xiwen <forbidden405@outlook.com> */
+
+#include <debug.h>
+
+#include <libfdt.h>
+
+#include <lk2nd/hw/gpio.h>
+
+#include "device.h"
+
+#define LK2ND_MAX_LEDS 16
+
+enum led_state {
+	LED_STATE_UNKNOWN = -1,
+	LED_STATE_ON,
+	LED_STATE_OFF,
+	LED_STATE_KEEP,
+};
+
+static int lk2nd_leds_init(const void *dtb, int node)
+{
+	int i = 0, subnode, len;
+	struct gpiol_desc desc;
+	const char *state;
+	enum led_state estate = LED_STATE_UNKNOWN;
+
+	dprintf(SPEW, " Initializing LEDs\n");
+	dprintf(SPEW, " | label | code  | gpio        |\n");
```

Comment:
this header doesn't match the table output below ("code" is useless for leds)

Reply:
removed code, added state